[BUG] dask_cudf `test_groupby_cumulative[null-cumsum-True]` intermittently failing with CUDA_ERROR_ILLEGAL_ADDRESS

I was seeing a failure with a PR I have up, but I am seeing the same failure in other PRs, so I am reporting this issue.

When `test_groupby_cumulative` (https://github.com/rapidsai/cudf/blob/branch-23.06/python/dask_cudf/dask_cudf/tests/test_groupby.py#L93) includes nulls (`pdf` is given the "nulls" param), I see the test failing often:

https://github.com/rapidsai/cudf/actions/runs/4949862614/jobs/8853844720
https://github.com/rapidsai/cudf/actions/runs/4942045478/jobs/8835788977
https://github.com/rapidsai/cudf/actions/runs/4949538839/jobs/8853094643 

I do not see the failure for the `non_null` variant of the test.

Here's a repro:

```
./build.sh cudf libcudf dask_cudf
compute-sanitizer pytest python/dask_cudf/build/lib/dask_cudf/tests/test_groupby.py -s -vvv -k test_groupby_cumulative\ and\ null-cumsum-True
```

I have run this with compute-sanitizer with symbols and line numbers, and it repros every time there with an invalid global read of 8 bytes:

```
========= Invalid __global__ read of size 8 bytes 
=========     at 0x9c0 in /home/abellina/work/cudf/cpp/build/_deps/thrust-src/thrust/system/cuda/detail/transform.h:119:void thrust::cuda_cub::__transform::unary_transform_f<thrust::permutation_iterator<const long *, cudf::detail::input_indexalator>, long *, thrust::cuda_cub::__transform::no_stencil_tag, thrust::cuda_cub::identity, thrust::cuda_cub::__transform::always_true_predicate>::operator ()<long>(T1)
=========     by thread (235,0,0) in block (2,0,0)
=========     Address 0x7f4e424cb530 is out of bounds
=========     and is 113 bytes after the nearest allocation at 0x7f4e424c1800 of size 40,128 bytes 
=========     Device Frame:/home/abellina/work/cudf/cpp/build/_deps/thrust-src/thrust/system/cuda/detail/parallel_for.h:97:void thrust::cuda_cub::__parallel_for::ParallelForAgent<thrust::cuda_cub::__transform::unary_transform_f<thrust::permutation_iterator<const long *, cudf::detail::input_indexalator>, long *, thrust::cuda_cub::__transform::no_stencil_tag, thrust::cuda_cub::identity, thrust::cuda_cub::__transform::always_true_predicate>, long>::consume_tile<(bool)0>(thrust::cuda_cub::__transform::unary_transform_f<thrust::permutation_iterator<const long *, cudf::detail::input_indexalator>, long *, thrust::cuda_cub::__transform::no_stencil_tag, thrust::cuda_cub::identity, thrust::cuda_cub::__transform::always_true_predicate>, long, int) [0x270]
=========     Device Frame:/home/abellina/work/cudf/cpp/build/_deps/thrust-src/thrust/system/cuda/detail/parallel_for.h:118:thrust::cuda_cub::__parallel_for::ParallelForAgent<thrust::cuda_cub::__transform::unary_transform_f<thrust::permutation_iterator<const long *, cudf::detail::input_indexalator>, long *, thrust::cuda_cub::__transform::no_stencil_tag, thrust::cuda_cub::identity, thrust::cuda_cub::__transform::always_true_predicate>, long>::entry(thrust::cuda_cub::__transform::unary_transform_f<thrust::permutation_iterator<const long *, cudf::detail::input_indexalator>, long *, thrust::cuda_cub::__transform::no_stencil_tag, thrust::cuda_cub::identity, thrust::cuda_cub::__transform::always_true_predicate>, long, char *) [0x1a0]
=========     Device Frame:/home/abellina/work/cudf/cpp/build/_deps/thrust-src/thrust/system/cuda/detail/core/agent_launcher.h:76:void thrust::cuda_cub::core::_kernel_agent<thrust::cuda_cub::__parallel_for::ParallelForAgent<thrust::cuda_cub::__transform::unary_transform_f<thrust::permutation_iterator<const long *, cudf::detail::input_indexalator>, long *, thrust::cuda_cub::__transform::no_stencil_tag, thrust::cuda_cub::identity, thrust::cuda_cub::__transform::always_true_predicate>, long>, thrust::cuda_cub::__transform::unary_transform_f<thrust::permutation_iterator<const long *, cudf::detail::input_indexalator>, long *, thrust::cuda_cub::__transform::no_stencil_tag, thrust::cuda_cub::identity, thrust::cuda_cub::__transform::always_true_predicate>, long>(T2, T3) [0x10]
=========     Saved host backtrace up to driver entry point at kernel launch time
=========     Host Frame: [0x305c18]
=========                in /usr/lib/x86_64-linux-gnu/libcuda.so.1
=========     Host Frame: [0x1488c]
=========                in /home/abellina/micromamba/envs/cudf_dev/lib/python3.10/site-packages/cupy_backends/cuda/api/../../../../../libcudart.so.11.0
=========     Host Frame:cudaLaunchKernel [0x6c318]
=========                in /home/abellina/micromamba/envs/cudf_dev/lib/python3.10/site-packages/cupy_backends/cuda/api/../../../../../libcudart.so.11.0
=========     Host Frame:/usr/local/cuda/include/cuda_runtime.h:215:cudaError cudaLaunchKernel<char>(char const*, dim3, dim3, void**, unsigned long, CUstream_st*) [0x44c1d64]
=========                in /home/abellina/micromamba/envs/cudf_dev/lib/libcudf.so
=========     Host Frame:/tmp/tmpxft_00007bd3_00000000-6_gather.cudafe1.stub.c:774:__device_stub__ZN6thrust8cuda_cub4core13_kernel_agentINS0_14__parallel_for16ParallelForAgentINS0_11__transform17unary_transform_fINS_20permutation_iteratorIPKlN4cudf6detail17input_indexalatorEEEPlNS5_14no_stencil_tagENS0_8identityENS5_21always_true_predicateEEElEESI_lEEvT0_T1_(thrust::cuda_cub::__transform::unary_transform_f<thrust::permutation_iterator<long const*, cudf::detail::input_indexalator>, long*, thrust::cuda_cub::__transform::no_stencil_tag, thrust::cuda_cub::identity, thrust::cuda_cub::__transform::always_true_predicate>&, long) [0x44b8bd4]
=========                in /home/abellina/micromamba/envs/cudf_dev/lib/libcudf.so
=========     Host Frame:/tmp/tmpxft_00007bd3_00000000-6_gather.cudafe1.stub.c:778:void thrust::cuda_cub::core::__wrapper__device_stub__kernel_agent<thrust::cuda_cub::__parallel_for::ParallelForAgent<thrust::cuda_cub::__transform::unary_transform_f<thrust::permutation_iterator<long const*, cudf::detail::input_indexalator>, long*, thrust::cuda_cub::__transform::no_stencil_tag, thrust::cuda_cub::identity, thrust::cuda_cub::__transform::always_true_predicate>, long>, thrust::cuda_cub::__transform::unary_transform_f<thrust::permutation_iterator<long const*, cudf::detail::input_indexalator>, long*, thrust::cuda_cub::__transform::no_stencil_tag, thrust::cuda_cub::identity, thrust::cuda_cub::__transform::always_true_predicate>, long>(thrust::cuda_cub::__transform::unary_transform_f<thrust::permutation_iterator<long const*, cudf::detail::input_indexalator>, long*, thrust::cuda_cub::__transform::no_stencil_tag, thrust::cuda_cub::identity, thrust::cuda_cub::__transform::always_true_predicate>&, long&) [0x44b8c00]
=========                in /home/abellina/micromamba/envs/cudf_dev/lib/libcudf.so
=========     Host Frame:/home/abellina/work/cudf/cpp/build/_deps/thrust-src/thrust/system/cuda/detail/core/agent_launcher.h:316:void thrust::cuda_cub::core::_kernel_agent<thrust::cuda_cub::__parallel_for::ParallelForAgent<thrust::cuda_cub::__transform::unary_transform_f<thrust::permutation_iterator<long const*, cudf::detail::input_indexalator>, long*, thrust::cuda_cub::__transform::no_stencil_tag, thrust::cuda_cub::identity, thrust::cuda_cub::__transform::always_true_predicate>, long>, thrust::cuda_cub::__transform::unary_transform_f<thrust::permutation_iterator<long const*, cudf::detail::input_indexalator>, long*, thrust::cuda_cub::__transform::no_stencil_tag, thrust::cuda_cub::identity, thrust::cuda_cub::__transform::always_true_predicate>, long>(thrust::cuda_cub::__transform::unary_transform_f<thrust::permutation_iterator<long const*, cudf::detail::input_indexalator>, long*, thrust::cuda_cub::__transform::no_stencil_tag, thrust::cuda_cub::identity, thrust::cuda_cub::__transform::always_true_predicate>, long) [0x44c8781]
=========                in /home/abellina/micromamba/envs/cudf_dev/lib/libcudf.so
=========     Host Frame:/home/abellina/work/cudf/cpp/build/_deps/thrust-src/thrust/system/cuda/detail/core/triple_chevron_launch.h:78:cudaError thrust::cuda_cub::launcher::triple_chevron::doit_host<void (*)(thrust::cuda_cub::__transform::unary_transform_f<thrust::permutation_iterator<long const*, cudf::detail::input_indexalator>, long*, thrust::cuda_cub::__transform::no_stencil_tag, thrust::cuda_cub::identity, thrust::cuda_cub::__transform::always_true_predicate>, long), thrust::cuda_cub::__transform::unary_transform_f<thrust::permutation_iterator<long const*, cudf::detail::input_indexalator>, long*, thrust::cuda_cub::__transform::no_stencil_tag, thrust::cuda_cub::identity, thrust::cuda_cub::__transform::always_true_predicate>, long>(void (*)(thrust::cuda_cub::__transform::unary_transform_f<thrust::permutation_iterator<long const*, cudf::detail::input_indexalator>, long*, thrust::cuda_cub::__transform::no_stencil_tag, thrust::cuda_cub::identity, thrust::cuda_cub::__transform::always_true_predicate>, long), thrust::cuda_cub::__transform::unary_transform_f<thrust::permutation_iterator<long const*, cudf::detail::input_indexalator>, long*, thrust::cuda_cub::__transform::no_stencil_tag, thrust::cuda_cub::identity, thrust::cuda_cub::__transform::always_true_predicate>, long) const [0x456a644]
=========                in /home/abellina/micromamba/envs/cudf_dev/lib/libcudf.so
=========     Host Frame:/home/abellina/work/cudf/cpp/build/_deps/thrust-src/thrust/system/cuda/detail/parallel_for.h:163:void thrust::cuda_cub::parallel_for<thrust::detail::execute_with_allocator<rmm::mr::thrust_allocator<char>, thrust::cuda_cub::execute_on_stream_nosync_base>, thrust::cuda_cub::__transform::unary_transform_f<thrust::permutation_iterator<long const*, cudf::detail::input_indexalator>, long*, thrust::cuda_cub::__transform::no_stencil_tag, thrust::cuda_cub::identity, thrust::cuda_cub::__transform::always_true_predicate>, long>(thrust::cuda_cub::execution_policy<thrust::detail::execute_with_allocator<rmm::mr::thrust_allocator<char>, thrust::cuda_cub::execute_on_stream_nosync_base> >&, thrust::cuda_cub::__transform::unary_transform_f<thrust::permutation_iterator<long const*, cudf::detail::input_indexalator>, long*, thrust::cuda_cub::__transform::no_stencil_tag, thrust::cuda_cub::identity, thrust::cuda_cub::__transform::always_true_predicate>, long) [0x4556157]
=========                in /home/abellina/micromamba/envs/cudf_dev/lib/libcudf.so
=========     Host Frame:/home/abellina/work/cudf/cpp/build/_deps/thrust-src/thrust/system/cuda/detail/gather.h:49:long* thrust::cuda_cub::gather<thrust::detail::execute_with_allocator<rmm::mr::thrust_allocator<char>, thrust::cuda_cub::execute_on_stream_nosync_base>, cudf::detail::input_indexalator, long const*, long*>(thrust::cuda_cub::execution_policy<thrust::detail::execute_with_allocator<rmm::mr::thrust_allocator<char>, thrust::cuda_cub::execute_on_stream_nosync_base> >&, cudf::detail::input_indexalator, cudf::detail::input_indexalator, long const*, long*) [0x450620e] 
=========                in /home/abellina/micromamba/envs/cudf_dev/lib/libcudf.so
=========     Host Frame:/home/abellina/work/cudf/cpp/build/_deps/thrust-src/thrust/detail/gather.inl:42:long* thrust::gather<thrust::detail::execute_with_allocator<rmm::mr::thrust_allocator<char>, thrust::cuda_cub::execute_on_stream_nosync_base>, cudf::detail::input_indexalator, long const*, long*>(thrust::detail::execution_policy_base<thrust::detail::execute_with_allocator<rmm::mr::thrust_allocator<char>, thrust::cuda_cub::execute_on_stream_nosync_base> > const&, cudf::detail::input_indexalator, cudf::detail::input_indexalator, long const*, long*) [0x44e75fd] 
=========                in /home/abellina/micromamba/envs/cudf_dev/lib/libcudf.so
=========     Host Frame:/home/abellina/work/cudf/cpp/include/cudf/detail/gather.cuh:139:void cudf::detail::gather_helper<long const*, long*, cudf::detail::input_indexalator>(long const*, int, long*, cudf::detail::input_indexalator, cudf::detail::input_indexalator, bool, rmm::cuda_stream_view) [0x44d9ec7]
=========                in /home/abellina/micromamba/envs/cudf_dev/lib/libcudf.so
=========     Host Frame:/home/abellina/work/cudf/cpp/include/cudf/detail/gather.cuh:226:std::unique_ptr<cudf::column, std::default_delete<cudf::column> > cudf::detail::column_gatherer_impl<long, void>::operator()<cudf::detail::input_indexalator>(cudf::column_view const&, cudf::detail::input_indexalator, cudf::detail::input_indexalator, bool, rmm::cuda_stream_view, rmm::mr::device_memory_resource*) [0x44d2e3a] 
=========                in /home/abellina/micromamba/envs/cudf_dev/lib/libcudf.so
=========     Host Frame:/home/abellina/work/cudf/cpp/include/cudf/detail/gather.cuh:183:std::unique_ptr<cudf::column, std::default_delete<cudf::column> > cudf::detail::column_gatherer::operator()<long, cudf::detail::input_indexalator>(cudf::column_view const&, cudf::detail::input_indexalator, cudf::detail::input_indexalator, bool, rmm::cuda_stream_view, rmm::mr::device_memory_resource*) [0x44ca6a9] 
=========                in /home/abellina/micromamba/envs/cudf_dev/lib/libcudf.so
=========     Host Frame:/home/abellina/work/cudf/cpp/include/cudf/detail/gather.cuh:665:std::unique_ptr<cudf::table, std::default_delete<cudf::table> > cudf::detail::gather<cudf::detail::input_indexalator>(cudf::table_view const&, cudf::detail::input_indexalator, cudf::detail::input_indexalator, cudf::out_of_bounds_policy, rmm::cuda_stream_view, rmm::mr::device_memory_resource*) [0x44c3f34] 
=========                in /home/abellina/micromamba/envs/cudf_dev/lib/libcudf.so
=========     Host Frame:/home/abellina/work/cudf/cpp/src/copying/gather.cu:57:cudf::detail::gather(cudf::table_view const&, cudf::column_view const&, cudf::out_of_bounds_policy, cudf::detail::negative_index_policy, rmm::cuda_stream_view, rmm::mr::device_memory_resource*) [0x44b29b5]
=========                in /home/abellina/micromamba/envs/cudf_dev/lib/libcudf.so
=========     Host Frame:/home/abellina/work/cudf/cpp/src/groupby/groupby.cu:230:cudf::groupby::groupby::get_groups(cudf::table_view, rmm::mr::device_memory_resource*) [0x4a8e7ea]
```

The `null` option passed for me. The `non_null` failed. No memory issue showed up in my tests:
```
FAILED python/dask_cudf/build/lib/dask_cudf/tests/test_groupby.py::test_groupby_cumulative[non_null-cumsum-True] - ValueError: Length mismatch: Expected axis has 0 elements, new values have 10000 elements
FAILED python/dask_cudf/build/lib/dask_cudf/tests/test_groupby.py::test_groupby_cumulative[non_null-cumsum-False] - ValueError: Length mismatch: Expected axis has 0 elements, new values have 10000 elements
FAILED python/dask_cudf/build/lib/dask_cudf/tests/test_groupby.py::test_groupby_cumulative[non_null-cumcount-True] - ValueError: Length mismatch: Expected axis has 0 elements, new values have 10000 elements
FAILED python/dask_cudf/build/lib/dask_cudf/tests/test_groupby.py::test_groupby_cumulative[non_null-cumcount-False] - ValueError: Length mismatch: Expected axis has 0 elements, new values have 10000 elements
```

Example stacktrace:
```
______________________________________________________________________________________________________________________________ test_groupby_cumulative[null-cumcount] ________________________________________________________________________________________________________________________________

aggregation = 'cumcount', pdf =        xx         x         y
0     NaN  0.349298 -0.307054
1     0.0 -0.748976       NaN
2     NaN       NaN -0.75043...417
9997  NaN       NaN  1.703114
9998  3.0       NaN -0.007891
9999  NaN       NaN       NaN

[10000 rows x 3 columns]

    @pytest.mark.parametrize("aggregation", ["cumcount"])
    def test_groupby_cumulative(aggregation, pdf):
        gdf = cudf.DataFrame.from_pandas(pdf)
        ddf = dask_cudf.from_cudf(gdf, npartitions=5)
    
        gdf_grouped = gdf.groupby("xx")
        ddf_grouped = ddf.groupby("xx")
    
        context = contextlib.nullcontext()
        with context:
>           a = getattr(gdf_grouped, aggregation)()

python/dask_cudf/dask_cudf/tests/test_groupby.py:91: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../../Softwares/miniconda3/envs/cudf_dev/lib/python3.10/site-packages/nvtx/nvtx.py:101: in inner
    result = func(*args, **kwargs)
../../../Softwares/miniconda3/envs/cudf_dev/lib/python3.10/site-packages/cudf/core/groupby/groupby.py:404: in cumcount
    .agg("cumcount")
../../../Softwares/miniconda3/envs/cudf_dev/lib/python3.10/site-packages/cudf/core/groupby/groupby.py:2291: in agg
    result = super().agg(func)
../../../Softwares/miniconda3/envs/cudf_dev/lib/python3.10/site-packages/nvtx/nvtx.py:101: in inner
    result = func(*args, **kwargs)
../../../Softwares/miniconda3/envs/cudf_dev/lib/python3.10/site-packages/cudf/core/groupby/groupby.py:576: in agg
    return self._mimic_pandas_order(result)
../../../Softwares/miniconda3/envs/cudf_dev/lib/python3.10/site-packages/cudf/core/groupby/groupby.py:2266: in _mimic_pandas_order
    result.index = self.obj.index
../../../Softwares/miniconda3/envs/cudf_dev/lib/python3.10/site-packages/cudf/core/dataframe.py:1093: in __setattr__
    super().__setattr__(key, col)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = Empty DataFrame
Columns: [None]
Index: [], value = RangeIndex(start=0, stop=10000, step=1)

    @index.setter
    def index(self, value):
        old_length = len(self)
        new_length = len(value)
    
        # A DataFrame with 0 columns can have an index of arbitrary length.
        if len(self._data) > 0 and new_length != old_length:
>           raise ValueError(
                f"Length mismatch: Expected axis has {old_length} elements, "
                f"new values have {len(value)} elements"
            )
E           ValueError: Length mismatch: Expected axis has 0 elements, new values have 10000 elements

../../../Softwares/miniconda3/envs/cudf_dev/lib/python3.10/site-packages/cudf/core/indexed_frame.py:536: ValueError
```

--- a/python/cudf/cudf/core/groupby/groupby.py
+++ b/python/cudf/cudf/core/groupby/groupby.py
@@ -2270,11 +2270,31 @@ class GroupBy(Serializable, Reducible, Scannable):
         """
         # TODO: copy metadata after this method is a common pattern, should
         # merge in this method.
-        _, order_cols, _ = self._groupby.groups(
-            [arange(0, result._data.nrows)]
+
+        # This function is used to reorder the results of scan-based
+        # groupbys which have the same output size as input size.
+        # However, if the grouping key has NAs and dropna=True, the
+        # result coming back from libcudf has null_count few rows than
+        # the input, so we must produce an ordering from the full
+        # input range.
+        _, (ordering,), _ = self._groupby.groups(
+            [arange(0, self.obj._data.nrows)]
         )
-        gather_map = order_cols[0].argsort()
-        result = result.take(gather_map)
+        if self._dropna and any(
+            c.has_nulls(include_nan=True) > 0
+            for c in self.grouping._key_columns
+        ):
+            # Scan aggregations with null/nan keys put nulls in the
+            # corresponding output rows in pandas, to do that here
+            # expand the result by reindexing.
+            ri = cudf.RangeIndex(0, self.obj._data.nrows)
+            result.index = cudf.Index(ordering)
+            # This reorders and expands
+            result = result.reindex(ri)
+        else:
+            # Just reorder according to the groupings
+            result = result.take(ordering.argsort())
+        # Now produce the actual index we first thought of
         result.index = self.obj.index
         return result
 

--- a/python/cudf/cudf/tests/test_groupby.py
+++ b/python/cudf/cudf/tests/test_groupby.py
@@ -2131,6 +2131,35 @@ def test_groupby_rank_fails():
         gdf.groupby(["a"]).rank(method="min", axis=1)
 
 
+@pytest.mark.parametrize(
+    "with_nan", [False, True], ids=["just-NA", "also-NaN"]
+)
+@pytest.mark.parametrize("dropna", [False, True], ids=["keepna", "dropna"])
+@pytest.mark.parametrize(
+    "duplicate_index", [False, True], ids=["rangeindex", "dupindex"]
+)
+def test_groupby_scan_null_keys(with_nan, dropna, duplicate_index):
+    key_col = [None, 1, 2, None, 3, None, 3, 1, None, 1]
+    if with_nan:
+        df = pd.DataFrame(
+            {"key": pd.Series(key_col, dtype="float32"), "value": range(10)}
+        )
+    else:
+        df = pd.DataFrame(
+            {"key": pd.Series(key_col, dtype="Int32"), "value": range(10)}
+        )
+
+    if duplicate_index:
+        # Non-default index with duplicates
+        df.index = [1, 2, 3, 1, 3, 2, 4, 1, 6, 10]
+
+    cdf = cudf.from_pandas(df)
+
+    expect = df.groupby("key", dropna=dropna).cumsum()
+    got = cdf.groupby("key", dropna=dropna).cumsum()
+    assert_eq(expect, got)
+
+
 def test_groupby_mix_agg_scan():
     err_msg = "Cannot perform both aggregation and scan in one operation"
     func = ["cumsum", "sum"]

--- a/python/dask_cudf/dask_cudf/tests/test_groupby.py
+++ b/python/dask_cudf/dask_cudf/tests/test_groupby.py
@@ -76,7 +76,18 @@ def test_groupby_basic(series, aggregation, pdf):
 
 # TODO: explore adding support with `.agg()`
 @pytest.mark.parametrize("series", [True, False])
-@pytest.mark.parametrize("aggregation", ["cumsum", "cumcount"])
+@pytest.mark.parametrize(
+    "aggregation",
+    [
+        "cumsum",
+        pytest.param(
+            "cumcount",
+            marks=pytest.mark.xfail(
+                reason="https://github.com/rapidsai/cudf/issues/13390"
+            ),
+        ),
+    ],
+)
 def test_groupby_cumulative(aggregation, pdf, series):
     gdf = cudf.DataFrame.from_pandas(pdf)
     ddf = dask_cudf.from_cudf(gdf, npartitions=5)

--- a/python/dask_cudf/dask_cudf/tests/test_groupby.py
+++ b/python/dask_cudf/dask_cudf/tests/test_groupby.py
@@ -1,7 +1,5 @@
 # Copyright (c) 2021-2023, NVIDIA CORPORATION.
 
-import contextlib
-
 import numpy as np
 import pandas as pd
 import pytest
@@ -90,17 +88,10 @@ def test_groupby_cumulative(aggregation, pdf, series):
         gdf_grouped = gdf_grouped.xx
         ddf_grouped = ddf_grouped.xx
 
-    if pdf.isna().sum().any():
-        # https://github.com/rapidsai/cudf/issues/12055
-        gdf_grouped = gdf.groupby("xx")
-        context = pytest.raises(ValueError)
-    else:
-        context = contextlib.nullcontext()
-    with context:
-        a = getattr(gdf_grouped, aggregation)()
-        b = getattr(ddf_grouped, aggregation)()
+    a = getattr(gdf_grouped, aggregation)()
+    b = getattr(ddf_grouped, aggregation)()
 
-        dd.assert_eq(a, b)
+    dd.assert_eq(a, b)
 
 
 @pytest.mark.parametrize("aggregation", OPTIMIZED_AGGS)
@@ -136,7 +127,6 @@ def test_groupby_agg(func, aggregation, pdf):
 
 @pytest.mark.parametrize("split_out", [1, 3])
 def test_groupby_agg_empty_partition(tmpdir, split_out):
-
     # Write random and empty cudf DataFrames
     # to two distinct files.
     df = cudf.datasets.randomdata()
@@ -496,7 +486,6 @@ def test_groupby_mean_sort_false():
 
 
 def test_groupby_reset_index_dtype():
-
     # Make sure int8 dtype is properly preserved
     # Through various cudf/dask_cudf ops
     #

